### PR TITLE
Type agnostic tab name and easy startup layout modification for editor_tabs

### PR DESCRIPTION
**Pull request by rewin123**
Part of #267 

This PR allows you to:
- use any type to set the name of the editor panel. 
- allows to support several types specifying the name at the same time 
- It is more convenient to specify the panels displayed when the editor is launched

Major changes:
- EditorTabName is no longer part of editor_tabs, but part of editor_ui
- Introduced the TabName trait, which should be released by the types that are used to name editor panels

```rust
trait TabName: Debug + Any {
    fn clear_background(&self) -> bool;
    fn title(&self) -> String;
}
```
- EditorTabName now implement TabName trait
- The api for registering new tabs has been changed accordingly

```rust
Was
app.editor_tab_by_trait(EditorTabName::GameView, GameViewTab::default());
Became
app.editor_tab_by_trait(GameViewTab::default());
```

```rust
Was
app.editor_tab(EditorTabName::Hierarchy, "Hierarchy".into(), show_hierarchy);
Bacame
app.editor_tab(EditorTabName::Hierarchy, show_hierarchy);
```

- Added api for setting the start layout of the editor
Usage:
```rust
/// Create layout group, which hide hard logic of egui_dock tree
app.init_layout_group::<DoublePanelGroup, _>();

/// Push tabs to group of layout
app.layout_push::<DoublePanelGroup, _, _>(DoublePanel::MainPanel, EditorTabName::GameView);
app.layout_push::<DoublePanelGroup, _, _>(DoublePanel::TopPanel, EditorTabName::Hierarchy);
app.layout_push::<DoublePanelGroup, _, _>(
    DoublePanel::BottomPanel,
    EditorTabName::Inspector,
);

/// We can push showed tab on front of list if we want
app.layout_push_front::<DoublePanelGroup, _, _>(
    DoublePanel::MainPanel,
    EditorTabName::Resource,
);
```

- Added useful example about how add new editor tabs
```rust
use bevy::prelude::*;
use space_editor::prelude::*;

/// This example shows how to create custom editor tabs
/// space_editor allows to create tabs by implementing trait EditorTab
/// or by using system based tabs

fn main() {
    App::default()
        .add_plugins(DefaultPlugins)
        .add_plugins(SpaceEditorPlugin)
        .add_systems(Startup, simple_editor_setup)
        // Add trait based tab
        .editor_tab_by_trait(TraitEditorTab)
        // Add system based tab
        .editor_tab(CustomTabName::SystemBased, system_tab)
        // Add trait based tab as first tab in Bottom panel
        .layout_push_front::<DoublePanelGroup, _, _>(
            DoublePanel::BottomPanel,
            CustomTabName::TraitBased,
        )
        // Add system based tab as first tab in Main panel
        .layout_push_front::<DoublePanelGroup, _, _>(
            DoublePanel::MainPanel,
            CustomTabName::SystemBased,
        )
        .run();
}

#[derive(Debug)]
/// A custom tab name
enum CustomTabName {
    /// A trait based tab
    TraitBased,
    /// A system based tab
    SystemBased,
}

impl TabName for CustomTabName {
    /// Set clear_background to true to clear background of the panel
    fn clear_background(&self) -> bool {
        true
    }

    /// Return title of the tab
    fn title(&self) -> String {
        match self {
            CustomTabName::TraitBased => String::from("Trait Based"),
            CustomTabName::SystemBased => String::from("System Based"),
        }
    }
}

#[derive(Resource)]
/// A struct that implements EditorTab trait
/// which allows to create custom tabs in the editor
struct TraitEditorTab;

impl EditorTab for TraitEditorTab {
    /// This function is called when tab needs to be rendered
    /// ui is bevy_egui::egui::Ui and it allows to build ui inside the tab
    fn ui(
        &mut self,
        ui: &mut space_prefab::prelude::ext::bevy_inspector_egui::egui::Ui,
        _commands: &mut Commands,
        _world: &mut World,
    ) {
        ui.label("Trait Based");
    }

    /// Return name of the tab
    fn tab_name(&self) -> TabNameHolder {
        CustomTabName::TraitBased.into()
    }
}

/// This function is a system that will be called every frame and will construct tab ui
fn system_tab(mut commands: Commands, mut ui: NonSendMut<EditorUiRef>) {
    let ui = &mut ui.0;

    ui.label("System Based");

    // If button is clicked spawn a new entity with
    // SpatialBundle and Name components
    if ui.button("Add").clicked() {
        commands.spawn((
            SpatialBundle::default(),
            PrefabMarker,
            Name::new("New Entity".to_string()),
        ));
    }
}
```

Next steps in next PRs:
- Move colors and sizing modules out of editor_tabs crate

### CODE REVIEW

**Comment by rewin123** [comment]
image
Test without features CI failed with this error, weird.....

**Comment by naomijub** [comment]
> image Test without features CI failed with this error, weird.....

Might be a linker problem on CI. Maybe we could run CI on windows or be sure to have it correctly setup for linux https://github.com/bevyengine/bevy/blob/main/docs/linux_dependencies.md 

But I see this is passing now

**Comment by rewin123** [comment]
> I would always recommend adding UI screenshots when changing stuff that touch the UI, besides that, this is a pretty good change and clearly documented. Nice work

UI shouldn't have changed, just the api

File: crates/editor_tabs/src/start_layout.rs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+pub trait GroupLayout<G>: StartLayout {
+    /// Add tab to group to end
+    fn push<N: TabName>(&mut self, group: G, tab: N);
+    /// Add tab to group to start
+    fn push_front<N: TabName>(&mut self, group: G, tab: N);
+}
+
+pub enum DoublePanel {
+    TopPanel,
```
Comment (naomijub) [inline crates/editor_tabs/src/start_layout.rs:21]: Maybe call them TopLeftPanel and BottomLeftPanel? first time I read them I felt confused with the menutoolbars

**Review by naomijub — APPROVED** [review]

File: crates/editor_tabs/src/start_layout.rs (hunk 1 above)
Comment (rewin123) [reply crates/editor_tabs/src/start_layout.rs:21]: Maybe then call TopLeft and BottomLeft?  I love reducing words count

**Pull request merged**